api/envoy/service/ext_proc/v3alpha/external_processor.proto

```diff
@@ -74,29 +74,37 @@ message ProcessingRequest {
     option (validate.required) = true;
 
     // Information about the HTTP request headers, as well as peer info and additional
-    // properties. If "response_required" is set, the server must send back a
+    // properties. If "async_mode" is set, the server must send back a
     // HeaderResponse message, an ImmediateResponse message, or close the stream.
     HttpHeaders request_headers = 2;
 
     // Information about the HTTP response headers, as well as peer info and additional
-    // properties. If "response_required" is set, the server must send back a
+    // properties. If "async_mode" is set, the server must send back a
     // HeaderResponse message or close the stream.
     HttpHeaders response_headers = 3;
 
-    // A chunk of the HTTP request body. If "response_required" is set, the server must send back
+    // A chunk of the HTTP request body. If "async_mode" is set, the server must send back
     // a BodyResponse message, an ImmediateResponse message, or close the stream.
     HttpBody request_body = 4;
 
-    // A chunk of the HTTP request body. If "response_required" is set, the server must send back
+    // A chunk of the HTTP request body. If "async_mode" is set, the server must send back
     // a BodyResponse message or close the stream.
     HttpBody response_body = 5;
 
-    // The HTTP trailers for the request path. If "response_required" is set, the server
-    // must send back a TrailerResponse message or close the stream.
+    // The HTTP trailers for the request path. If "async_mode" is set, the server
+    // must send back a TrailerResponse message or close the stream. This message is
+    // sent if the trailers processing mode is set to "SEND". If there are no trailers
+    // on the original downstream request, then this message will only be sent
+    // (with empty trailers waiting to be populated) if the processing mode is set
```

Comment:
Is there are any value in distinguishing between an Envoy client that sends an empty trailers frame and no trailers frame?

Reply:
The only value IMO is in making it clearer to the server developer what to expect. I think it's easy to check whether the trailers frame will be empty, but if someone wants to write a filter that, for example, checksums the response body and adds the result in a trailer, then they need to be able to add trailers when they don't exist.

However, I could also see the need for a mode that says, basically, "only send me trailers if they already exist." But that's adding yet more logic here so I was hoping to keep it simpler for now.